Mlem/App/Views/Root/Tabs/Feeds/SectionIndexTitles.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  SectionIndexTitles.swift
+//  Mlem
+//
+//  Created by mormaer on 13/08/2023.
+//
+//
+
+import Dependencies
+import SwiftUI
+
+struct SectionIndexTitles: View {
+    struct Section: Identifiable {
+        let label: String
+        var systemImage: String?
+        var id: String { label }
+    }
+    
+    let proxy: ScrollViewProxy
+    let sections: [Section]
+    @GestureState private var dragLocation: CGPoint = .zero
+
+    // Track which sidebar label we picked last to we
+    // only haptic when selecting a new one
+    @State var lastSelectedLabel: String = ""
+
+    var body: some View {
+        VStack {
+            ForEach(sections) { communitySection in
+                sectionTitle(for: communitySection)
+                    .frame(width: 12, height: 6)
+            }
+        }
+        .overlay {
+            GeometryReader { geo in
+                // Color.clear doesn't register gestures (presumably because it never gets drawn), so we fake it
+                Color.black
+                    .opacity(0.00000000001)
```

Comment:
We use this little trick a couple different places--we should put it in a `Color` extension

Reply:
I found an alternative to this:

```swift
Color.clear
    .contentShape(.rect)
```